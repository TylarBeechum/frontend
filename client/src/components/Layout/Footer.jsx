import React from 'react';
import styled from 'styled-components';

const Footer = () => {
  return (
    <FooterContainer>
      <span>Â© Stampd 2019</span>
    </FooterContainer>
  );
};
const FooterContainer = styled.div`
  text-align: center;
  position: fixed;
  bottom: 0;
  width: 100%;
  color: #333333;
  background: #f8f8f8;
  padding-bottom: 10px;
`;

export default Footer;

// ${props => props.theme.global.colors['neutral-3']}
// ${props => props.theme.global.colors['accent-3']}
