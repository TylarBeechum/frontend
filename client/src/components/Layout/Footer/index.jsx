import React from 'react';
import styled from 'styled-components';

const Footer = props => {
  return (
    <FooterContainer>
      <span>stampd2019</span>
    </FooterContainer>
  );
};
const FooterContainer = styled.div`
  text-align: center;
  border-top: 1px solid #333;
  position: fixed;
  bottom: 0;
  width: 100%;
  color: white;
  background-color: #3aecfc;
  border-top: 2px solid #82fdff;
`;

export default Footer;

// ${props => props.theme.global.colors['neutral-3']}
//${props => props.theme.global.colors['accent-3']}
