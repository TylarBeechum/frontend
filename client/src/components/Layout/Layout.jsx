import React from 'react';
import { NavBar } from './navBar';
import { Footer } from './footer';

const Layout = ({ children }) => {
  return (
    <div>
      <NavBar />
      {children}
      <Footer />
    </div>
  );
};

export default Layout;
