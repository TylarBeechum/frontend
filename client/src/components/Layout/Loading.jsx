import React from 'react';
import Loader from 'react-loader-spinner';
import styled from 'styled-components';

const Loading = () => {
  return (
    <LoadingContainer>
      <Loader type="RevolvingDot" color="#7D4CDB" height={100} width={100} />
    </LoadingContainer>
  );
};

const LoadingContainer = styled.div`
  display: flex;
  min-height: 100vh;
  background: #f8f8f8;
  align-items: center;
  justify-content: center;
`;

export default Loading;
