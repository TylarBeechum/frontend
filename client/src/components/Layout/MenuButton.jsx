import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const MenuButton = ({ setShown, isShown }) => {
  const [isOpen, setIsOpen] = useState(false);
  const setMenu = e => {
    e.preventDefault();
    setIsOpen(!isOpen);
    setShown(!isShown);
  };
  return (
    <Styles>
      <div
        className={
          isOpen
            ? 'hamburger-menu menu-open clickHamburgerMenu'
            : 'hamburger-menu'
        }
        onClick={e => setMenu(e)}
      >
        <span />
        <span />
        <span />
      </div>
    </Styles>
  );
};

MenuButton.propTypes = {
  setShown: PropTypes.func.isRequired,
  isShown: PropTypes.bool.isRequired
};

const Styles = styled.div`
  cursor: pointer;
  margin-right: 13px;

  .hamburger-menu {
    width: 33px;
    height: 33px;
    margin-right: 10px;
    position: relative;
    display: inline-block;

    span {
      background-color: white;
      position: absolute;
      border-radius: 2px;
      transition: 0.3s cubic-bezier(0.8, 0.5, 0.2, 1.4);
      width: 40px;
      height: 5px;
      transition-duration: 0.4s;
    }
    span:nth-child(1) {
      top: 0px;
      left: 0px;
    }
    span:nth-child(2) {
      top: 14px;
      left: 0px;
    }
    span:nth-child(3) {
      bottom: 0px;
      left: 0px;
    }
  }
  .menu-open {
    z-index: 3;
  }
  .clickHamburgerMenu span:nth-child(1) {
    transform: rotate(45deg);
    top: 14px;
    width: 43px;
  }
  .clickHamburgerMenu span:nth-child(2) {
    transform: scale(0);
  }
  .clickHamburgerMenu span:nth-child(3) {
    transform: rotate(-45deg);
    top: 14px;
    width: 43px;
  }
`;

export default MenuButton;
