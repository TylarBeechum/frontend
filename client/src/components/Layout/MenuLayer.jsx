import React from 'react';
import styled, { keyframes, css } from 'styled-components';
import { Box, Button, Layer, Text } from 'grommet';
import PropTypes from 'prop-types';
import { useStateValue } from 'react-conflux';
import { globalContext } from '../../store/reducers/globalReducer';
import c from '../../store/constants';

const MenuLayer = ({ isShown, onClose, history }) => {
  const [{ user }] = useStateValue(globalContext);
  const navRoute = (e, route) => {
    e.preventDefault();
    history.push(route);
    onClose();
  };
  const menuArray = user.roleId === '2' ? c.schoolRoutes : c.studentRoutes;
  return (
    <MenuBar
      className={isShown && 'menu-bar__shown'}
      position="left"
      full="vertical"
      plain
      onClickOutside={() => onClose()}
      isShown={isShown}
    >
      <Box pad="medium" background="brand" fill="vertical">
        {menuArray.map(navItem => (
          <Button
            key={navItem.name}
            onClick={e => navRoute(e, navItem.route)}
            hoverIndicator={{ background: 'red' }}
          >
            <Box pad={{ horizontal: 'medium', vertical: 'small' }}>
              <Text className="navbar__link" size="large">
                {navItem.name}
              </Text>
            </Box>
          </Button>
        ))}
      </Box>
    </MenuBar>
  );
};

MenuLayer.propTypes = {
  isShown: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const menuSlide = keyframes`
  0% {
    margin: 70px 0 0 -275px;
  }
  100% {
    margin: 70px 0 0;
  }
`;

const MenuBar = styled(Layer)`
  padding: 0;
  width: 275px;
  height: calc(100vh - 70px);
  animation: none;
  ${props =>
    props.isShown &&
    css`
      animation: 0.3s ${menuSlide} forwards;
    `}

  .navbar__link {
    transition: opacity 0.3s;

    :hover {
      opacity: 0.6;
    }
  }
`;

export default MenuLayer;
