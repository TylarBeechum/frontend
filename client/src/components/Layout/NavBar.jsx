import React, { useState } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useStateValue } from 'react-conflux';
import { SecondaryButton } from '../../styles/themes';

import { useAuth0 } from '../../auth/authWrapper';
import { globalContext, LOGOUT } from '../../store/reducers/globalReducer';
import MenuLayer from './MenuLayer';
import stampdLogoWhite from '../../images/stampd_full_white.png';
import MenuButton from './MenuButton';

function NavBar({ history }) {
  const [{ onboarded }, globalDispatch] = useStateValue(globalContext);
  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();
  const [isShown, setShown] = useState(false);
  const [loading, setLoading] = useState(true);

  const toggleOpen = () => {
    setShown(!isShown);
    setLoading(false);
  };

  const handleLogout = () => {
    globalDispatch({
      type: LOGOUT,
      payload: false
    });
    localStorage.clear();
    setShown(false);
    setLoading(false);
    logout();
  };

  return (
    <NavContainter>
      <nav>
        <div className="left-section">
          {onboarded && (
            <MenuButton
              className="hamburger"
              setShown={setShown}
              isShown={isShown}
              onClose={toggleOpen}
              setLoading={setLoading}
              loading={loading}
            />
          )}
          <div className="logo">
            <Link to={onboarded ? '/dashboard' : '/'}>
              <img src={stampdLogoWhite} alt="Stampd logo" draggable="false" />
            </Link>
          </div>
        </div>
        <RightSection>
          <LinksSection>
            {!isAuthenticated && (
              <p>
                <Link to="/about">About Us</Link>
              </p>
            )}
            {!isAuthenticated && (
              <p>
                <Link to="/contact">Contact</Link>
              </p>
            )}
          </LinksSection>
          {!isAuthenticated ? (
            <div className="button__container">
              <NavBtn
                a11yTitle="Login"
                type="button"
                onClick={() => loginWithRedirect({})}
                label="Login"
              />
            </div>
          ) : (
            <div className="button__container">
              {/* <img src="" alt="avatar" /> */}
              <NavBtn
                a11yTitle="Logout"
                type="button"
                onClick={handleLogout}
                label="Logout"
              />
            </div>
          )}
        </RightSection>
      </nav>
      {isShown && (
        <MenuLayer
          loading={loading}
          setLoading={setLoading}
          toggleOpen={toggleOpen}
          isShown={isShown}
          setShown={setShown}
          history={history}
        />
      )}
    </NavContainter>
  );
}

NavBar.propTypes = {
  history: PropTypes.object.isRequired // eslint-disable-line react/forbid-prop-types
};

const NavContainter = styled.div`
  display: flex;
  width: 100%;
  height: 70px;
  position: fixed;
  padding: 0 3%;
  justify-content: space-between;
  background-color: ${props => props.theme.global.colors.brand};
  align-items: center;
  top: 0;
  -webkit-box-shadow: 0px 0px 15px -2px rgba(0, 0, 0, 0.46);
  -moz-box-shadow: 0px 0px 15px -2px rgba(0, 0, 0, 0.46);
  box-shadow: 0px 0px 15px -2px rgba(0, 0, 0, 0.46);
  z-index: 50;

  nav {
    margin: 0 auto;
    width: 100%;
    height: 70px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    position: relative;

    .left-section {
      display: flex;
      align-items: center;

      .logo {
        width: 113px;
        height: auto;

        @media (max-width: 500px) {
          display: none;
        }

        a {
          display: flex;
          align-content: center;
        }
      }

      .hamburger {
        cursor: pointer;
      }
    }
  }
`;

const RightSection = styled.div`
  display: flex;
  align-items: center;
  flex-direction: row;
  justify-content: space-between;
  width: 300px;

  p {
    color: white;
  }

  .button__container {
    right: 0;
    display: flex;
    justify-content: flex-end;
  }
`;

const NavBtn = styled(SecondaryButton)`
  border: 2px solid white /*${props => props.theme.global.colors['neutral-2']}*/;
  color: white /*${props => props.theme.global.colors['neutral-2']}*/;
  transition: background 0.3s, color 0.3s;

  :hover {
    color: ${({ theme }) => theme.global.colors.brand};
    background: ${({ theme }) => theme.global.colors.navbarHoverBg};
  }
`;

const LinksSection = styled.div`
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  margin-right: 3%;
  p {
    margin-right: 7px;
    a {
      color: white;
      :hover {
        text-decoration: underline;
      }
    }
  }
`;

export default NavBar;
