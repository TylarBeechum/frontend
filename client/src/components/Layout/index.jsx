import React from 'react';
import { Grommet } from 'grommet';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import NavBar from './NavBar';
import Footer from './Footer';
import { theme } from '../../styles/themes';

// GlobalStyles will go here

const Layout = ({ children }) => {
  return (
    <>
      <Grommet theme={theme}>
        <Container>
          <NavBar />
          {children}
          <Footer />
        </Container>
      </Grommet>
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

const Container = styled.main`
  width: 100%;
  position: relative;
  .brand-background {
    background: ${props => props.theme.global.colors.brand};
  }
  .status-ok {
    background: ${props => props.theme.global.colors['status-ok']};
  }
`;

export default Layout;
