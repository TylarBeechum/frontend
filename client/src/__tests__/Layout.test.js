/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import Layout from '../components/Layout';

describe('<Layout />', () => {
  describe('should render the Layout component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(
        <Layout history={{ location: { pathname: '/' } }}>
          <p>Test</p>
        </Layout>
      );
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(
        <Layout history={{ location: { pathname: '/' } }}>
          <p>Test</p>
        </Layout>
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should render the navbar to the screen successfully', () => {
    it('tests that the navbar has been rendered successfully', () => {
      const { getByLabelText } = renderWithRouterAndProviders(
        <Layout history={{ location: { pathname: '/' } }}>
          <p>Test</p>
        </Layout>
      );
      const loginBtn = getByLabelText(/login/i);
      expect(loginBtn).toBeDefined();
    });
  });

  describe('should render the footer to the screen successfully', () => {
    it('tests that the footer has been rendered successfully', () => {
      const { getByText } = renderWithRouterAndProviders(
        <Layout history={{ location: { pathname: '/' } }}>
          <p>Test</p>
        </Layout>
      );
      const copyright = getByText(/Â© Stampd 2019/i);
      expect(copyright).toBeDefined();
    });
  });
});
