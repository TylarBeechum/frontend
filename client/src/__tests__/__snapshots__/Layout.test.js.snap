// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <div
    className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
  >
    <main
      className="sc-dnqmqq cwhyFX"
    >
      <div
        className="sc-bZQynM CZNzZ"
      >
        <nav>
          <div>
            <svg
              aria-label="Menu"
              className="hamburger StyledIcon-ofa7kd-0 bBBqGu"
              data-testid="hamburger"
              onClick={[Function]}
              viewBox="0 0 24 24"
            >
              <path
                d="M2,19 L22,19 M2,5 L22,5 M2,12 L22,12"
                fill="none"
                stroke="#000"
                strokeWidth="2"
              />
            </svg>
            <div>
              <a
                href="/dashboard"
                onClick={[Function]}
              >
                <img
                  alt="Stampd logo"
                  draggable="false"
                  src="stampd_full_white.svg"
                />
              </a>
            </div>
          </div>
          <div
            className="button__container"
          >
            <button
              aria-label="Logout"
              className="sc-bwzfXH sc-gzVnrw dhncQw StyledButton-sc-323bzc-0 kHrfdt"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="button"
            >
              Logout
            </button>
          </div>
        </nav>
      </div>
      <p>
        Test
      </p>
      <div
        className="sc-htoDjs eBjQTr"
      >
        <p>
          Â© Stampd 2019
        </p>
      </div>
    </main>
  </div>
</div>
`;
